app/User.php

```diff
@@ -46,4 +46,40 @@ public function centre()
     {
         return $this->belongsTo('App\Centre');
     }
+
+    /**
+     * Get the relevant centres for this User, accounting for it's role
+     *
+     * @return \Illuminate\Database\Eloquent\Collection|\Illuminate\Support\Collection|static[]
+     */
+    public function relevantCentres()
+    {
+        // default to empty collection
+        $centres = collect();
+        switch ($this->role) {
+            case "FM_User":
+                // Just get all centres
+                $centres = Centre::all();
+                break;
+            case "CC_User":
+                // If we have one, get our centre's neighbors
+                if (!is_null($this->centre)) {
+                    $centres = $this->centre->neighbors;
+                }
+                break;
+        }
+        return $centres;
+    }
+
+    // ToDo : Principle of Responsibility - Does this live here or on the Centre model?
```

Comment:
I think it's OK here. Otherwise, you'd pass the User to Centre and ask - 'are you one of my centres?' ? I think it makes more sense next to the function that gets all User's centres.

Reply:
Cool, ta.